### ISSUE TITLE
harbor-go-client: a go CLI tool for harbor

### ISSUE THREAD

**Issue by moooofly** [goharbor/harbor#5703]
After Harbor [Community Call yesterday](https://github.com/goharbor/harbor/wiki/Community-Conference-Calls),  I am very glad to introduce you a CLI tool named [`harbor-go-client`](https://github.com/moooofly/harbor-go-client) (briefly mentioned in #1192, #5085, #5165 and #5285).

`harbor-go-client` is a command line interface (CLI) tool based on project harbor's REST APIs to try to facilitate CI/CD process.

The implementation of `harbor-go-client` is based on harbor v1.5.0-d59c257e and swagger api version 1.4.0.

Currently, `harbor-go-client` has already supported most of the swapper APIs.

In the meanwhile, `harbor-go-client` support  two special subcommands to do some cleanups by retention policy:

- `rp_tags`: Do tags deletion on repositories according to retention policy.
- `rp_repos`: Do soft deletion on repositories according to retention policy (prompt user performing a GC after that).

There are two common use cases:

- run `rp_tags` as an analysis tool to watch tags status and do deletion as your wish.
- debug online issues , especially by checking logs, such as when replication error happens between endpoints. [harbor/issues/3795](https://github.com/goharbor/harbor/issues/3795), [distribution/issues/2625](https://github.com/docker/distribution/issues/2625), [harbor/issues/5166](https://github.com/goharbor/harbor/issues/5166)

Based on the community call we had yesterday,  there are some things I can keep improving: 

- Add remaining APIs.
- Add golang test codes.
- refactor with cmd framework like `viper`, etc. 
- optimize log output format more human-friendly.

/cc @steven-zou

**Comment by steven-zou** [comment]
@moooofly 

Some suggestions for the go cli:
- Follow the normal go command style (use subcommand)
- Build based on the matured command framework like cobra

**Comment by moooofly** [comment]
@steven-zou 

I see, that makes sense to me!

**Comment by moooofly** [comment]
Hi @clouderati 

At the beginning, this project was specifically designed based on the needs of my company. Now, as an independent community contributor, I plan to make it a "general purpose" CLI tool for Harbor as my long-term goal.

**Comment by steven-zou** [comment]
As @moooofly is already working on implementing this proposal, assign this issue to @moooofly.

**Comment by steven-zou** [comment]
@moooofly 

 We're planning to use md file as a proposal content format and use PR raised in [goharbor/harbor](https://github.com/goharbor/community) for reviewing.

  Would u please follow the guideline below to move your proposal to there?

  ```
  git clone https://github.com/goharbor/community
  git checkout -b YOUR_BRANCH
  # add md file with meaningful-words_proposal.md name under `proposals/new` folder
  git commit
  # raise PR to request merge proposal
  # reviewers can append comments to the PR
  ```

**Comment by moooofly** [comment]
@steven-zou 

ok, I will prepare a proposal this weekend.

**Comment by moooofly** [comment]
@steven-zou 

The proposal is available at [goharbor/community#6](https://github.com/goharbor/community/pull/6) now.
